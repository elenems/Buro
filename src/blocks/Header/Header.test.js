import { shallow, mount } from "enzyme";
import React from "react";
import HeaderDefault, { Header } from "./Header.jsx";
import Button from "@material-ui/core/Button";
import { BrowserRouter } from "react-router-dom";

describe("Header", () => {
  it("Renders", () => {
    const wrapper = shallow(<Header />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Redirects to /add", () => {
    const wrapper = mount(
      <BrowserRouter>
        <HeaderDefault />
      </BrowserRouter>
    );
    wrapper.find(Button).simulate("click");
    expect(wrapper.find("Header").prop("history").location.pathname).toBe(
      "/add"
    );
  });
});
