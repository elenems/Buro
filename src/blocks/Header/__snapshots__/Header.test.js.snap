// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders 1`] = `
"<WithStyles(ForwardRef(Grid)) classes={{...}} item={true} xs={12} md={6}>
  <header>
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Button)) onClick={[Function: moveToAddPage]} variant=\\"contained\\" color=\\"primary\\" endIcon={{...}}>
        Розмістити
      </WithStyles(ForwardRef(Button))>
      <SearchBar />
    </Styled(MuiBox)>
  </header>
</WithStyles(ForwardRef(Grid))>"
`;
